### PR TITLE
Add Spanish locales

### PR DESCRIPTION
**Pull request by iwiznia**
Please, let's not focus too much on the translations themselves, we can iterate on them later.

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/152156

### Tests
- Go to the JS console > Application > Local Storage and set a new key `preferredLocale` to `"es-ES"` (including the quotes)
- Play around the app and check texts are correctly translated to spanish
- Delete the preferredLocale if you don't want e.cash to remain in spanish forever

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web
image
image
image
image

### CODE REVIEW

**Comment by shawnborton** [comment]
For this one:

image

Could we either add a word or remove a word so that we don't have the hanging orphan on the second line?

**Comment by iwiznia** [comment]
> Could we either add a word or remove a word so that we don't have the hanging orphan on the second line?

I don't think we should, mainly because these are not final copy, it's just a test so that we can see the whole I18n project works... also it is possible that in some languages the copy will be hard to fit in one line, so we need to learn to live with it... it does not look that bad to me.

**Comment by pecanoro** [comment]
For that one maybe we can simply change it to "¡Cuéntanos algo sobre tí, nos encantaría conocerte!"

**Comment by iwiznia** [comment]
Alright changed it

File: src/languages/es.js [hunk 1]
```diff
@@ -1,3 +1,279 @@
...
+        priorityMode: 'Modo Prioridad',
+    },
+    signInPage: {
+        expensifyDotCash: 'Expensify.cash',
+        expensifyIsOpenSource: 'Expensify.cash es open source',
+        theCode: 'el código',
+        openJobs: 'trabajos disponibles',
+    },
+    termsOfUse: {
+        phrase1: 'Al ingresar, estas aceptando los',
```
Comment (pecanoro) [inline src/languages/es.js:216]:
```suggestion
        phrase1: 'Al usar Expensify.cash, estás aceptando los',
```

File: src/languages/es.js [hunk 2]
```diff
@@ -1,3 +1,279 @@
...
+        phrase2: 'términos de servicio',
+        phrase3: 'y',
+        phrase4: 'política de privacidad',
+        phrase5: '. El envío de dinero es brindado por Expensify Payments LLC (NMLS ID:2017010) de conformidad con sus',
+        phrase6: 'licencias',
+    },
+    passwordForm: {
+        pleaseFillOutAllFields: 'Por favor completa todos los campos',
+        forgot: 'Te has olvidado?',
```
Comment (pecanoro) [inline src/languages/es.js:225]:
```suggestion
        forgot: '¿Te has olvidado?',
```

File: src/languages/es.js [hunk 3]
```diff
@@ -1,3 +1,279 @@
...
+        twoFactorCode: 'Autenticación de 2 factores',
```
Comment (pecanoro) [inline src/languages/es.js:226]: `Autentificación` sounds better than `Autenticación`, but maybe it's just me

File: src/languages/es.js [hunk 4]
```diff
@@ -1,3 +1,279 @@
...
+        requiredWhen2FAEnabled: 'Obligatorio cuando A2F esta habilitado',
+    },
+    loginForm: {
+        pleaseEnterEmailOrPhoneNumber: 'Por favor escribe un email o número de teleéfono',
```
Comment (pecanoro) [inline src/languages/es.js:230]:
```suggestion
        pleaseEnterEmailOrPhoneNumber: 'Por favor escribe un email o número de teléfono',
```

File: src/languages/es.js [hunk 5]
```diff
@@ -1,3 +1,279 @@
...
+        phoneOrEmail: 'Teleéfono o Email',
```
Comment (pecanoro) [inline src/languages/es.js:231]:
```suggestion
        phoneOrEmail: 'Número de Teléfono o Email',
```

File: src/languages/es.js [hunk 6]
```diff
@@ -1,3 +1,279 @@
...
+        enterYourPhoneOrEmail: 'Escribe tu teleéfono o email:',
```
Comment (pecanoro) [inline src/languages/es.js:232]:
```suggestion
        enterYourPhoneOrEmail: 'Escribe tu número de teléfono o email:',
```

File: src/languages/es.js [hunk 7]
```diff
@@ -1,3 +1,279 @@
...
+    },
+    resendValidationForm: {
+        linkHasBeenResent: 'El enlace se ha reenviado',
+        weSentYouMagicSignInLink: 'Te hemos enviado un enlace - simplemente haz click en él para conectarte!',
```
Comment (pecanoro) [inline src/languages/es.js:236]:
```suggestion
        weSentYouMagicSignInLink: '¡Te hemos enviado un enlace - simplemente haz click en él para conectarte!',
```

File: src/languages/es.js [hunk 8]
```diff
@@ -1,3 +1,279 @@
...
+        getMeOutOfHere: 'Sácame de aquí',
+    },
+    setPasswordPage: {
+        passwordCannotBeBlank: 'La contraseña no puede estar vacía',
+        enterPassword: 'Escribe una contraseña',
+        setPassword: 'Configura tu Contraseña',
+    },
+    addBankAccountPage: {
+        enterPassword: 'Escribe una contraseña',
+        addBankAccount: 'Agregar una cuenta de banco',
```
Comment (pecanoro) [inline src/languages/es.js:256]:
```suggestion
        addBankAccount: 'Agregar una cuenta bancaria',
```

File: src/languages/es.js [hunk 9]
```diff
@@ -1,3 +1,279 @@
...
+        alreadyAdded: 'Esta cuenta ya ha sido agregada.',
+        selectAccount: 'Selecciona una cuenta:',
+    },
+    attachmentView: {
+        unknownFilename: 'Archivo desconocido',
+    },
+    pronouns: {
+        heHimHis: 'El',
```
Comment (pecanoro) [inline src/languages/es.js:264]:
```suggestion
        heHimHis: 'Él',
```

File: src/languages/es.js [hunk 10]
```diff
@@ -1,3 +1,279 @@
...
+        selfSelect: 'Personalízalo',
+        callMeByMyName: 'Llámame por mi nombre',
+    },
+    cameraPermissionsNotGranted: 'No has habilitado los permisos para acceder a la cámara',
+    messages: {
+        noPhoneNumber: 'Por favor escribe un número de teléfono que incluya el código de país e.g +447814266907',
+        maxParticipantsReached: 'Has llegado al número máximo de participantes para un grupo.',
+    },
+    session: {
+        offlineMessage: 'Parece que no estás conectado a internet. Comprueba to conexión e inténtalo de nuevo?',
```
Comment (pecanoro) [inline src/languages/es.js:277]:
```suggestion
        offlineMessage: 'Parece que no estás conectado a internet. Comprueba tu conexión e inténtalo de nuevo.',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: src/languages/es.js [hunk 11]
```diff
@@ -1,3 +1,279 @@
...
+        updatePrompt: 'Esta disponible una nueva version de Expensify.cash.\nActualiza ahora or reinicia la aplicación luego para recibir la última version.',
+    },
+    iOUConfirmationList: {
+        whoPaid: 'QUIÉN PAGO?',
+        whoWasThere: 'QUIÉN ESTABA?',
+        whatsItFor: 'PARA QUÉ ES?',
+    },
+    iOUCurrencySelection: {
+        selectCurrency: 'Selecciona una moneda',
+        allCurrencies: 'TODAS LAS MONEDAS',
```
[the remark this answers is not in the document]
Comment (iwiznia) [reply src/languages/es.js:59]: Moneda is more common IMO

File: src/languages/es.js [hunk 12]
```diff
@@ -1,3 +1,279 @@
...
+        buttonSearch: 'Buscar',
+        buttonMySettings: 'Mi configuración',
+        fabNewChat: 'Nuevo Chat',
+    },
+    iou: {
+        amount: 'Importe',
+        participants: 'Participantes',
+        confirm: 'Confirmar',
+        splitBill: 'Dividir Factura',
+        requestMoney: 'Pedir Dinero',
```
[the remark this answers is not in the document]
Comment (iwiznia) [reply src/languages/es.js:111]: Solicitar importe no me dice mucho...

**Comment by iwiznia** [comment]
Updated

File: src/languages/es.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (pecanoro) [reply src/languages/es.js:111]: `Pedir dinero` sounds like begging money to me 😂

File: src/languages/es.js [hunk 13]
```diff
@@ -1,3 +1,279 @@
...
+        phrase3: 'y',
+        phrase4: 'política de privacidad',
+        phrase5: '. El envío de dinero es brindado por Expensify Payments LLC (NMLS ID:2017010) de conformidad con sus',
+        phrase6: 'licencias',
+    },
+    passwordForm: {
+        pleaseFillOutAllFields: 'Por favor completa todos los campos',
+        forgot: '¿Te has olvidado?',
+        twoFactorCode: 'Autenticación de 2 factores',
+        requiredWhen2FAEnabled: 'Obligatorio cuando A2F esta habilitado',
```
Comment (pecanoro) [inline src/languages/es.js:227]:
```suggestion
        requiredWhen2FAEnabled: 'Obligatorio cuando A2F está habilitado',
```

File: src/languages/es.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (iwiznia) [reply src/languages/es.js:111]: 🤷 we can change this later

**Comment by iwiznia** [comment]
Updated

**Review by pecanoro — APPROVED** [review]

**Pull request merged**